src/neuroconv/tools/nwb_helpers/_configuration_models/_base_dataset_io.py

```diff
@@ -67,105 +84,55 @@ def _infer_dtype(dataset: Union[h5py.Dataset, zarr.Array]) -> np.dtype:
     return data_type
 
 
-class DatasetInfo(BaseModel):
```

Comment:
This is the biggest change as per our previous discussions on the matter

Prior to Pydantic 2 the entire model was either mutable or immutable; post Pydantic 2 individual fields can be frozen

So instead of having an entire nested class inside the other models (the `DatasetIOConfiguration`s) these values are now unpacked and attributed directly to the dataset models